src/main/java/frc/robot/subsystems/endgame/EndgameSubsystem.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.subsystems.endgame;
+
+import com.revrobotics.CANSparkMax;
+
+import org.photonvision.targeting.PhotonPipelineResult;
+
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+
+import frc.robot.Constants.EndgameConstants;
+import frc.robot.subsystems.endgame.EndgameSubsystem;
+import frc.robot.subsystems.endgame.EndgameIO.EndgameIOInputs;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class EndgameSubsystem extends SubsystemBase{
+    private PIDController endgameController = new PIDController(0, 0, 0);
+    private double endgamekP = 0.03;
+    private double endgamekI = 0.0;
+    private double endgamekD = 0.0;
+    EndgameIO endgameIO;
+    EndgameIOInputsAutoLogged endgameInputs = new EndgameIOInputsAutoLogged();
+
+    // SmartDashboard
+    // SmartDashboard 
+    private double leftEndgameMotorPower = 0.5;
+    private double rightEndgameMotorPower = 0.5;
+    private double endgameGoalPosition;
+
+    public EndgameSubsystem(EndgameIO endgameIO) {
+        this.endgameIO = endgameIO;
+    }
+
+
+    public void setEndgamePosition() {
+        //endgamekP = SmartDashboard.getNumber("endgame kP", 0.03);
+        //endgamekI = SmartDashboard.getNumber("endgame kI", 0.0);
+        //endgamekD = SmartDashboard.getNumber("endgame kD", 0.0);
+
+    }
+
+    public void endgameGoUp(){
+        endgameGoalPosition = EndgameConstants.endgameUp;
+    }
+    public void endgameGoDown(){
+        endgameGoalPosition = EndgameConstants.endgameDown;
+    }
+
+    public void endgameControl() {
+        double leftOutput = endgameController.calculate(endgameInputs.encoderLeftPosition, endgameGoalPosition);
+        double rightOutput = endgameController.calculate(endgameInputs.encoderRightPosition, endgameGoalPosition);
+        endgameIO.setEndgameMotorPower(leftOutput, rightOutput);
+        double amps = endgameInputs.endgameAmps;
+    }
+  @Override
+  public void periodic(){
+    endgameIO.updateInputs(endgameInputs);
+    endgameController.setPID(endgamekP, endgamekI, endgamekD);
+    SmartDashboard.putNumber("left endgame motor power", leftEndgameMotorPower);
+    SmartDashboard.putNumber("right endgame motor power", rightEndgameMotorPower);
```

Comment:
These should probably go in the `endgameControl` method and use left/right Output instead. I'm not sure the leftEndgameMotorPower and rightEndgameMotorPower variables are used anywhere